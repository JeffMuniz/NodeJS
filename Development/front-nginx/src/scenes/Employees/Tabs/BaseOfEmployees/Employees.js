import React from "react";
import { func, number, shape, arrayOf, oneOf } from "prop-types";

import { requestStatus } from "@enums";

import { WithPagination } from "@base";

import { toNumberMask } from "@utils";

import {
  ValueLabelRow,
  ValueLabelDiv,
  ValueLabelTextStyled,
} from "./Employees.styles";
import EmployeesList from "./EmployeesList/EmployeesList";

const statuses = Object.values(requestStatus);

const fieldsToFilterBy = [
  {
    key: "cpf",
    description: "nº do CPF",
    optionDescription: "CPF",
    mask: "cpf",
  },
  { key: "name", description: "Nome" },
];

const fieldsToFilterByStatus = [
  { key: 1, description: "Ativos", label: "status" },
  { key: 2, description: "Inativos", label: "status" },
  { key: 0, description: "Todos", label: "status" },
];

export const EmployeesMain = ({
  employeesCount,
  employeeList,
  callback,
  employeeListRequestStatus,
  selectedCompany,
  page,
}) => (
  <div>
    <ValueLabelRow>
      <ValueLabelDiv>
        <ValueLabelTextStyled
          id="e_total_employees"
          title={toNumberMask(employeesCount.totalEmployees || 0)}
          subtitle="funcionários"
        />
        <ValueLabelTextStyled
          id="e_active_employees"
          title={toNumberMask(employeesCount.activeEmployees || 0)}
          subtitle="ativos"
        />
        <ValueLabelTextStyled
          id="e_active_employees"
          title={toNumberMask(employeesCount.inactiveEmployees || 0)}
          subtitle="inativos"
        />
      </ValueLabelDiv>
    </ValueLabelRow>
    <WithPagination
      data={employeeList}
      fieldsToFilterBy={fieldsToFilterBy}
      showSearchInput
      showFilterDropdownGeneric
      fieldsToFilterDropdownGeneric={fieldsToFilterByStatus}
      callback={callback}
      itemsPerPage={10}
      status={employeeListRequestStatus}
      selectedCompany={selectedCompany}
      showLoading
      currentPage={page}
      render={props => <EmployeesList data={employeeList.content} {...props} />}
    />
  </div>
);

EmployeesMain.propTypes = {
  employeesCount: shape({}).isRequired,
  employeeList: shape({
    totalItems: number,
    content: arrayOf(shape({})),
  }).isRequired,
  callback: func.isRequired,
  employeeListRequestStatus: oneOf(statuses).isRequired,
  selectedCompany: shape({
    id: number,
  }),
  page: number.isRequired,
};

EmployeesMain.defaultProps = {
  selectedCompany: {
    id: null,
    name: "",
    cnpj: "",
    description: "",
    value: "",
  },
};
export default EmployeesMain;
