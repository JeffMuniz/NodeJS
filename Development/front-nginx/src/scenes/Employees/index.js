import React from "react";
import { isNil } from "lodash";
import { shape } from "prop-types";
import { Tabs } from "@common";
import { ContainerWrapper } from "@base";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

// import EmployeesGroupAccess from "src/modules/EmployeesGroupAccess/EmployeesGroupAccess";

import {
  BaseOfEmployees,
  CardTracking,
  BulkChargeback,
  EmployeesStatus,
  BalanceTransfer,
} from "./Tabs";

const Finances = ({ location: { state }, selectedCompany }) => {
  const listIncludesCnpj = ["71632754000102", "30798783000161"];

  const tabs = [
    {
      id: "employee-base-tab",
      label: "Base de Funcionários",
      component: BaseOfEmployees,
    },
    {
      id: "employee-card-tracking-tab",
      label: "Tracking de Cartões",
      component: CardTracking,
    },
    {
      id: "employee-chargeback-tab",
      label: "Estorno",
      component: BulkChargeback,
    },
    {
      id: "employee-change-status-tab",
      label: "Ativar e Inativar",
      component: EmployeesStatus,
    },
    {
      id: "employee-balance-transfer",
      label: "Transferência de Saldo",
      component: BalanceTransfer,
      isHidden: !listIncludesCnpj.includes(selectedCompany.cnpj),
    },
  ];

  return (
    <ContainerWrapper title="Gerenciar Funcionários" isBiggerTitle>
      <Tabs
        activeTab={
          isNil(state) || !isNil(state.page)
            ? "Base de Funcionários"
            : state.route.toString()
        }
      >
        {tabs
          .filter(el => !el.isHidden)
          .map(({ id, label, component: Component }) => (
            <div id={id} label={label}>
              <Component />
            </div>
          ))}
      </Tabs>
    </ContainerWrapper>
  );
};

Finances.propTypes = {
  location: shape({}).isRequired,
  selectedCompany: shape({}).isRequired,
};

export const mapStateToProps = ({ selectedCompanyTree }) => ({
  selectedCompany: selectedCompanyTree.selectedCompany,
});

export default withRouter(connect(mapStateToProps)(Finances));
