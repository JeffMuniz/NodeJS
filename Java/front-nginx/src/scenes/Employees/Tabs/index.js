export { default as BaseOfEmployees } from "./BaseOfEmployees/Employees.container";
export { default as CardTracking } from "./CardTracking/CardTracking.container";
export { default as BulkChargeback } from "./BulkChargeback";
export { default as EmployeesStatus } from "./EmployeesStatus";
export { default as BalanceTransfer } from "./BalanceTransfer/BalanceTransfer.container";
