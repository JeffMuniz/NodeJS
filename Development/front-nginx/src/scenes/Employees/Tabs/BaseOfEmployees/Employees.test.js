import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import Employees from "./Employees";

jest.mock("@base", () => ({
  ContainerWrapper: "ContainerWrapper",
  WithPagination: "WithPagination",
  TableCol: "TableCol",
  TableHeaderCol: "TableHeaderCol",
  OrderBy: "OrderComponent",
}));
jest.mock("./EmployeesList/EmployeesList", () => "EmployeesList");
jest.mock("./Employees.container", () => "EmployeesMain");

jest.mock("@common", () => ({
  Tabs: "Tabs",
  ValueLabelText: "ValueLabelText",
}));

const defaultProps = {
  employeesCount: { totalEmployees: 100, activeEmployees: 100 },
  employeeListRequestStatus: "none",
  updateEmployeesData: jest.fn(),
  selectedCompany: {},
  onClick: jest.fn(),
  employeeList: {},
  callback: jest.fn(),
  page: 1,
};

describe("Employees - Scene", () => {
  it("Should render it", () => {
    // given
    const renderer = new ShallowRenderer();

    // when
    renderer.render(<Employees {...defaultProps} />);
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });
});
