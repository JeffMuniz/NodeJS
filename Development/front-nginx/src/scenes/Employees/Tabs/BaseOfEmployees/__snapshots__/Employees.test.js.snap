// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employees - Scene Should render it 1`] = `
<div>
  <styled.div>
    <styled.div>
      <Styled(ValueLabelText)
        id="e_total_employees"
        subtitle="funcionários"
        title="100"
      />
      <Styled(ValueLabelText)
        id="e_active_employees"
        subtitle="ativos"
        title="100"
      />
      <Styled(ValueLabelText)
        id="e_active_employees"
        subtitle="inativos"
        title="0"
      />
    </styled.div>
  </styled.div>
  <WithPagination
    callback={[MockFunction]}
    currentPage={1}
    data={Object {}}
    fieldsToFilterBy={
      Array [
        Object {
          "description": "nº do CPF",
          "key": "cpf",
          "mask": "cpf",
          "optionDescription": "CPF",
        },
        Object {
          "description": "Nome",
          "key": "name",
        },
      ]
    }
    fieldsToFilterDropdownGeneric={
      Array [
        Object {
          "description": "Ativos",
          "key": 1,
          "label": "status",
        },
        Object {
          "description": "Inativos",
          "key": 2,
          "label": "status",
        },
        Object {
          "description": "Todos",
          "key": 0,
          "label": "status",
        },
      ]
    }
    itemsPerPage={10}
    render={[Function]}
    selectedCompany={Object {}}
    showFilterDropdownGeneric={true}
    showLoading={true}
    showSearchInput={true}
    status="none"
  />
</div>
`;
